R/model.R

```diff
@@ -1024,7 +1029,23 @@ optimize_method <- function(data = NULL,
                             algorithm = NULL,
                             init_alpha = NULL,
                             iter = NULL,
-                            sig_figs = NULL) {
+                            sig_figs = NULL,
+                            threads = NULL) {
+  checkmate::assert_integerish(threads, lower = 1, len = 1, null.ok = TRUE)
+  if (is.null(self$cpp_options()[["stan_threads"]])) {
+    if (!is.null(threads)) {
+      warning("'threads' is set but the model was not compiled with ",
+              "'cpp_options = list(stan_threads = TRUE)' so 'threads' will have no effect!",
```

Comment:
This doesn't have to be part of this PR, but I wonder if we should do something similar to the `pedantic` and `include_paths` arguments and just provide an argument to `compile()` to turn on threading instead of forcing the use of `cpp_options()`. Not totally sure, but something to ponder.

Reply:
Yeah, we had that originally and it's now deprecated. Let's tackle that in a separate issue/PR

Reply:
Oh yeah I forgot about that. Maybe we were right originally! But yeah we can deal with that later.